lib/emque/producing/message/message.rb

```diff
@@ -97,20 +97,25 @@ def to_json
         Oj.dump(data, :mode => :compat)
       end
 
-      def publish(publisher=Emque::Producing.publisher)
+      def publish(publisher=nil)
+        publisher ||= Emque::Producing.publisher
         log "publishing...", true
         if valid?
           log "valid...", true
           if Emque::Producing.configuration.publish_messages
             sent = publisher.publish(topic, message_type, to_json, partition_key)
             log "sent #{sent}"
-            if raise_on_failure? && !sent
-              raise MessagesNotSentError.new
-            end
+            raise MessagesNotSentError.new if raise_on_failure? && !sent
```

Comment:
Looking at this with fresh eyes, I don't think we need the `raise_on_failure?` check here anymore.